internal/indexer/linux/coalescer.go

```diff
@@ -170,3 +188,44 @@ func (c *Coalescer) Coalesce(ctx context.Context, artifacts []*indexer.LayerArti
 	}
 	return c.ir, nil
 }
+
+// shareRepos takes repository definition and share it with other layers
+// where repositories are missing
+func (c *Coalescer) shareRepos(ctx context.Context, artifacts []*indexer.LayerArtifacts) {
```

Comment:
@Allda This makes sense, but since this is RHEL specific I think we can take a slightly different approach.

We have a concept called "Ecosystem". an Ecosystem groups specific scanners with a particular Coalescer and runs a Coalescer on layers populated with artifacts *just* from those scanners. 

I'd like to see if we can do the following...

Take this coalescer and your modifications and create a "rhel/coalescer.go" file. It can be very similar to this one.
Create a new Ecosystem in "rhel/ecosystem.go". Define the components that are involved in this Ecosystem. Let's use this as an example: https://github.com/quay/claircore/blob/master/alpine/ecosystem.go#L11

In your case Ecosystem.PackageScanners would be the RPM package scanner, Ecosystem.DistributionScanner will be the RHEL distro scanner, Ecosystem.RepositoryScanners will be the RHEL repo scanner you created in this PR, and Ecosystem.Coalescer will be the RHEL coalescer you are introducing as well. 

By doing this we are making sure that RHEL coalescing method is only used with the RPM pkg scanner, RHEL distro scanner, and RHEL coalescer ensuing the rhel specific logic will not conflict with any other coalescing happening for the generic linux use case.